### ISSUE TITLE
C# / official or unofficial logo

### ISSUE THREAD

**Issue by kytrinyx** [exercism/meta#27]
As described in https://github.com/exercism/meta/issues/24, this weekend we're designing proper icons for each of the language tracks on Exercism.

In order to help speed us through this work, it would be helpful to know whether or not the language has a logo, whether official or unofficial, what that logo is, and what the usage rights of it are.

If you know things, awesome! Drop that knowledge below. If you don't but want to help do research, we will be very grateful.

Here's a template for you to fill out:

```
### Official Logo

### Unofficial Logo

### Link to Logo

### License, Usage Rights
```

For each heading, please list what you know or find out. If something is irrelevant due to something listed in an earlier heading, say `n/a` (not applicable).

We did some research a while back, and might have some useful information at the bottom of the [README.md](https://github.com/exercism/csharp/blob/master/README.md). (Look for "C# icon")

You can see whatever icon we've put together for the track in the [img directory](https://github.com/exercism/csharp/tree/master/img) of the track. Sometimes this is based on something official, sometimes it's just desperation made tangible. (You'll see why we're working with real designers.)

**Comment by ilya-khadykin** [comment]
### Official Logo
There is no official logo for C#, but people mentioned (in [this thread](https://stackoverflow.com/questions/2100109/is-there-an-official-logo-of-c) for example) that Microsoft .NET or Microsoft Visual C# ones could be used instead, not sure if it's relevant

Microsoft Visual C# logo (not sure about license):
![](https://media.licdn.com/mpr/mpr/AAEAAQAAAAAAAAkYAAAAJDIwMjA0NzUyLWVmNTAtNDhjNS05NThlLWRlN2I3YTZiMTczMA.png)

### Unofficial Logo
Some people use the general version (Public Domain) from https://en.wikipedia.org/wiki/File:C_Sharp_wordmark.svg:
![](https://upload.wikimedia.org/wikipedia/commons/thumb/0/0d/C_Sharp_wordmark.svg/464px-C_Sharp_wordmark.svg.png)

I've seen this one on Fedora website (not sure about license) which is referring to C/C++ versions - https://developer.fedoraproject.org/tech/languages/csharp/about.html
![](https://developer.fedoraproject.org/static/logo/csharp.png)
### Link to Logo
General one:
https://upload.wikimedia.org/wikipedia/commons/thumb/0/0d/C_Sharp_wordmark.svg/464px-C_Sharp_wordmark.svg.png

`librariesio/pictogram` (MIT license): https://github.com/librariesio/pictogram/blob/master/vendor/assets/images/c-sharp/c-sharp.png
### License, Usage Rights
General one (`C#`) consists of only of simple geometric shapes or text, and is therefore in the public domain.

Q: Is exercism.io using official logo?
A:  General version is used

**Comment by iHiD** [comment]
@m-a-ge That looks like it was a tricky one to make sense of. Thanks for digging in!

**Comment by ChrisMcKee** [comment]
You're free to use the purple C# logo btw https://chrismckee.co.uk/c-logo-for-stuff-and-stickers/
I basically recreated the C++ one years ago to contain the # and use VS colours at the time; Microsoft helped themselves to it so I made it clear the licence is go nuts 👍

**Comment by kytrinyx** [comment]
@ChrisMcKee Thank you! This is super helpful

**Comment by SylvainCorlay** [comment]
Found this thread as I was looking for a C# logo.

The logo proposed by @ChrisMcKee appears to be derived from the C++ logo. This may require authorization from the copyright holders of the C++ logo. There is some language about usage here: https://isocpp.org/home/terms-of-use

**Comment by ChrisMcKee** [comment]
@sylvaincorlay I did link to the post which links to the thread where I have precisely that permission.

If it makes you feel easier Microsoft didn't worry about using it in multiple places over the last few years either.

Thanks for taking the time to read before posting though...

**Comment by SylvainCorlay** [comment]
Thanks for your reply. There does not seem to be an open source license.

For another open source project logo (the cling c++ interpreter), we contacted isocpp about using a modified version of this logo and the answer was a clear no.

**Comment by ChrisMcKee** [comment]
@sylvaincorlay the issue there is with usage. There's a difference between utilising a logo for a similar purpose (c++ alt compiler) and cma different language. Ultimately Jeremy created it, and it was derivative of the C logo that was around.  If you look in the GitHub history he only committed his sources 2 years after I asked, and changed the license months later.

All the same I'll prod him again as there's nothing more fun than talking about  licenses that were altered and are unenforceable 😂
God I hate the internet at times

Ultimately this is a derivative work as it's original work based on another piece of work.
Within tedious trademark terms it would be unlikely to be confused with having anything to do with the iso cpp foundation ... Hell you'd struggle to find anyone who thought the c++ logo was linked to them.
There are no registered trademarks or copyrights around the works either.
Basically the licence text on their sites a load of bollocks conflating their various segmented names with the logo by failing to separate the two. 🙄

Just to cover other use 

![image](https://user-images.githubusercontent.com/83597/42414246-14817046-8229-11e8-98b6-911ae2975b0b.png)

**Comment by SylvainCorlay** [comment]
Thanks for the clarification. If you know the author and he clarifies the license, I would probably use it too!

Sorry for intruding here!

**Comment by ChrisMcKee** [comment]
Yeah @SylvainCorlay you can't revoke an open licence once given.

--------------------------------

TLDR for anyone looking at this sweating over getting dragged into licencing

* Jeremy created this and opensourced it.
* The Standard C++ Foundation eventually adopted it later
* The C# derivative was created during the term of the fully open-source licence
* You can't revoke a licence
* The licence was only changed on adoption and tranfer of repo in 2017

![image](https://user-images.githubusercontent.com/83597/42414300-33260a38-822a-11e8-9790-9b977f6645de.png)




```
As the copyright holder you can choose to re-license your code as you see fit. What you cannot do is revoke a license if it was given in perpetuity or before the term of the license expires. What does this mean?

You are able to license your code to different people using different licenses. You can even let them choose which license to use. You can sell the license, give it away, and whatever else you want to do with it.

Once someone has been granted a license then they are free to use the code in the terms of that license, though. Many open source licenses have provisions for the license being "perpetual", lasting forever. So, if you offer your code, for example, using the GPL license, then the people who use that license to use the code have perpetual rights to continue doing that. You cannot revoke those rights.

What you can do, is release a new version with a different license, then people using that new version will need to have new terms in place (or they could "fork" the project and continue using and maintaining the older, licensed version, them selves).

Summary: No, you cannot revoke a license, but you can change it for new users, and you can have multiple options for licensing.
```

**Comment by ChrisMcKee** [comment]
@SylvainCorlay sorry for the tone btw; I created this thing in 2013 and started printing stickers off (hence the PNG format) on redbubble 😆  It's like having to talk licencing about something that was opensource a life time ago 😜

**Comment by SylvainCorlay** [comment]
Thanks very much for digging this up!

I am now reassured that I will be able to do what I need with the logo!
